plugins/inputs/proxysql/proxysql.go

```diff
@@ -0,0 +1,552 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	// Global Variables may be gathered less often
+	if len(m.IntervalSlow) > 0 {
+		if uint32(time.Since(lastT).Seconds()) >= scanIntervalSlow {
+			err = m.gatherVariables(db, serv, acc, globalVariablesQuery, "proxysql_variables")
+			if err != nil {
+				return err
+			}
+			lastT = time.Now()
+		}
+	}
+
+	if m.GatherCommandsCounters {
+		err = m.gatherCommandsCounters(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	if m.GatherConnectionPool {
+		err = m.gatherConnectionPool(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	if m.GatherUsers {
+		err = m.gatherUsers(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	if m.GatherRules {
+		err = m.gatherRules(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	if m.GatherQueries {
+		err = m.gatherQueries(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	if m.GatherMemoryMetrics {
+		err = m.gatherVariables(db, serv, acc, memoryMetricsQuery, "proxysql_memory")
+		if err != nil {
+			return err
+		}
+	}
+
+	if m.GatherProcessList {
+		err = m.gatherProcessListStatuses(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// gatherGlobalVariables can be used to fetch all global variables from
+// MySQL environment.
+func (m *Proxysql) gatherVariables(db *sql.DB, serv string, acc telegraf.Accumulator, query string, measurement string) error {
+	// run query
+	rows, err := db.Query(query)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	var key string
+	var val sql.RawBytes
+
+	// parse DSN and save server tag
+	servtag := getDSNTag(serv)
+	tags := map[string]string{"server": servtag}
+	fields := make(map[string]interface{})
+	for rows.Next() {
+		if err := rows.Scan(&key, &val); err != nil {
+			return err
+		}
+		key = strings.ToLower(key)
+		// parse mysql version and put into field and tag
+		if strings.Contains(key, "version") {
+			fields[key] = string(val)
+			tags[key] = string(val)
+		}
+		if value, ok := parseValue(val); ok {
+			fields[key] = value
+		}
+		// Send 20 fields at a time
+		if len(fields) >= 20 {
+			acc.AddFields(measurement, fields, tags)
+			fields = make(map[string]interface{})
+		}
+	}
+	// Send any remaining fields
+	if len(fields) > 0 {
+		acc.AddFields(measurement, fields, tags)
+	}
+	return nil
+}
+
+func (m *Proxysql) gatherCommandsCounters(db *sql.DB, serv string, acc telegraf.Accumulator) error {
+	// run query
+	rows, err := db.Query(commandsCounterQuery)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	var command string
+	var count uint32
+
+	// parse DSN and save server tag
+	servtag := getDSNTag(serv)
+	tags := map[string]string{"server": servtag}
+	fields := make(map[string]interface{})
+	for rows.Next() {
+		if err := rows.Scan(&command, &count); err != nil {
+			return err
+		}
+		command = strings.ToLower(command)
+		fields[newNamespace("commands", command)] = count
```

Comment:
This seems like it would make more sense as a tag, can you show some examples of this measurement in line protocol format?